tapdb/universe_stats_test.go

```diff
@@ -247,81 +247,247 @@ func TestUniverseQuerySyncStatsSorting(t *testing.T) {
 
 	// sortCheck is used to generate an IsSorted func bound to the
 	// response, for each sort type below.
-	type sortCheck func([]universe.AssetSyncSnapshot) func(i, j int) bool
+	type sortCheck func([]universe.AssetSyncSnapshot, universe.SortDirection) func(i, j int) bool
 
 	// With the events above logged, we'll now make sure we can properly
 	// retrieve each of the events by their sorted order.
 	var tests = []struct {
 		name         string
 		sortType     universe.SyncStatsSort
+		direction    universe.SortDirection
 		isSortedFunc sortCheck
 	}{
 		{
-			name:     "asset name",
-			sortType: universe.SortByAssetName,
+			name:      "asset name sort ascending",
+			sortType:  universe.SortByAssetName,
+			direction: universe.SortAscending,
 			isSortedFunc: func(s []universe.AssetSyncSnapshot,
-			) func(i, j int) bool {
+				d universe.SortDirection) func(i, j int) bool {
 
 				return func(i, j int) bool {
-					return s[i].AssetName < s[j].AssetName
+					if d == universe.SortAscending {
```

Comment:
We could extract all of these individual `isSortedFunc` into a single one, now that they have become a bit more complicated.
For example:
```
	isSorted := func(s []universe.AssetSyncSnapshot,
		t universe.SyncStatsSort,
		d universe.SortDirection) func(i, j int) bool {

		asc := d == universe.SortAscending
		desc := d == universe.SortDescending
			
		return func(i, j int) bool {
			switch {
			case t == universe.SortByAssetName && asc:
				return s[i].AssetName < s[j].AssetName
			case t == universe.SortByAssetName && desc:
				return s[i].AssetName > s[j].AssetName
				
				...
			}
		}
	}
```